### PR TITLE
pre-commit: introduce pre-commit and fix linters

### PR DESCRIPTION
**Pull request by schuellerf**
not sure if the settings of those pre-commit checks/linters are perfect but this commit would be a start to avoid having the round-trip with failing github tests, hopefully

### CODE REVIEW

File: test/scripts/boot-image [hunk 1]
```diff
@@ -66,18 +78,34 @@ def ensure_uncompressed(filepath):
 def cmd_boot_aws(arch, image_name, privkey, pubkey, image_path, script_cmd):
     # pylint: disable=too-many-arguments
     aws_config = get_aws_config()
-    cmd = ["go", "run", "./cmd/boot-aws", "run",
-           "--access-key-id", aws_config["key_id"],
```
Comment (mvo5) [inline test/scripts/boot-image:70]: Uh, let's please not do this change here, this makes the `cmd` much harder to read IMHO. It's a bit silly that `ruff` cannot be configured to do better here (or can it)?

Comment (schuellerf) [reply test/scripts/boot-image:70]: I saw that too - I'll switch this to draft and dig deeper

**Comment by schuellerf** [comment]
> A bit of a meta comment, would it make sense to introduce PRs with one linter at a time? That would probably make both the review and the commit history a bit easier to read :)

Not sure if this would create an unnecessary high number of PRs - but the straight forward ones and `ruff` can be split, I guess🤔 😁

**Comment by schuellerf** [comment]
@mvo5 autopep8 seems to be ok, fixing lines that are too long but keeping others… please check

File: docs/developer/cmds.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 The following utilities, defined in the `cmd/` directory, are useful for
 development and testing. They **should not** be relied on for production
-purposes. In particular, command line options and default behaviour can change
+purposes. In particular, command line options and default behavior can change
```
Comment (kingsleyzissou) [inline docs/developer/cmds.md:5]: 😢 😭

**Review by kingsleyzissou — APPROVED** [review]

File: docs/developer/cmds.md (hunk 2 above)
Comment (schuellerf) [reply docs/developer/cmds.md:5]: Not sure if we should argue about
```
matrix:
    aspell:
      d: en_US
```
😉

Comment (kingsleyzissou) [reply docs/developer/cmds.md:5]: Not arguing, just hurts my soul a little 😆😭

**Pull request merged**